### PR TITLE
Add UID to Snapshot Tag

### PR DESCRIPTION
**Pull request by xing-yang**
Add UID from VolumeSnapshot Metadata as a tag to the snapshot created
by the plugin to make sure we can find the right snapshot through the
plugin corresponding to the VolumeSnapshot.

### CODE REVIEW

File: snapshot/pkg/cloudprovider/providers/openstack/openstack_snapshots.go [hunk 1]
```diff
@@ -234,12 +234,17 @@ func (os *OpenStack) FindSnapshot(tags map[string]string) ([]string, []string, e
...
 				if ok {
 					if tags[ctrlsnap.CloudSnapshotCreatedForVolumeSnapshotNameTag] == nameVal {
-						timeVal, ok := snapshot.Metadata[ctrlsnap.CloudSnapshotCreatedForVolumeSnapshotTimestampTag]
+						uidVal, ok := snapshot.Metadata[ctrlsnap.CloudSnapshotCreatedForVolumeSnapshotUIDTag]
 						if ok {
-							if tags[ctrlsnap.CloudSnapshotCreatedForVolumeSnapshotTimestampTag] == timeVal {
-								snapshotIDs = append(snapshotIDs, snapshot.ID)
-								statuses = append(statuses, snapshot.Status)
-								glog.Infof("Add snapshot [%#v].", snapshot)
+							if tags[ctrlsnap.CloudSnapshotCreatedForVolumeSnapshotUIDTag] == uidVal {
```
Comment (jingxu97) [inline snapshot/pkg/cloudprovider/providers/openstack/openstack_snapshots.go]: Does openstack support filter when listing snapshots? Use filter could avoid manually checking the meta data?

Comment (xing-yang) [reply snapshot/pkg/cloudprovider/providers/openstack/openstack_snapshots.go]: Metadata based filtering is supported by Cinder volume API microversion 3.21 and above. Currently the OpenStack Cloud Provider only supports V2.0.  I'll add a note to revisit this later.

Comment (xing-yang) [reply snapshot/pkg/cloudprovider/providers/openstack/openstack_snapshots.go]: A note is added.

**Pull request merged**